### DISCUSSION TITLE
2.8.11 更新日志

### DISCUSSION THREAD

**Discussion by LTCatt** [Meloong-Git/PCL#5237] [讨论]
### 更新亮点
- Mod 管理中添加了下载 Mod 选项（#4095，@212112111zwl）
    会自动跳转至下载页面，并筛选对应的版本和加载器（#4075，@JingHai-Lingyun）
- 如果没有 Mod，Mod 管理页面会提供下载和安装 Mod 的选项
    <img width="631" alt="image" src="https://github.com/user-attachments/assets/e5502e60-0dc2-49d4-8785-505c4f2f2c66">
- 允许正式版修改“打开启动器时显示 PCL 图标”设置（#4091，@lingyunawa）
- 新主页预设：杂志主页（#4926，由 @CreeperIsASpy 制作）
    你可以在 `设置 → 个性化 → 主页` 中选择主页预设！
    <img width="631" alt="image" src="https://github.com/user-attachments/assets/d508c022-6a26-4241-a0bd-f6384a4b5a61">
- 在 Mod 下载详情页添加了 MC 版本筛选（#5087，@xxtg666）
    <img width="631" alt="image" src="https://github.com/user-attachments/assets/50fa8dc8-2371-4858-b57a-d28c40df2f62">
- 开源协议更新：允许分发衍生作品
    现在，大家可以在 GitHub 下载 PCL 的源码，自己魔改 PCL 发布魔改版了！
    无论是整合包作者还是服主，只要满足协议要求，都可以修改 PCL 并且在自己的整合包中使用！
    详情：https://github.com/Hex-Dragon/PCL2/blob/main/LICENCE

### 其他更新内容
- 自动选择 Mod 下载文件夹时不再忽略 MC 的次要版本（#691，@WForst-Breeze）
    例如，下载 1.20.4 的 Mod 时不会再选择 1.20.1 的 Mod 文件夹
- 允许单独重置 JVM 参数头设置（#3284，@WForst-Breeze）
- 全盘搜索 Java 时不再搜索网络驱动器（#3705，@Pigeon0v0）
- 优化快照版输入更新密钥的流程（#4529，@Ad-closeNN）
- 允许某些隐藏主题在解锁后依然能够右键跳转（#4585，@Ad-closeNN）
- 内存优化进行中的提示时机有误（#4920，@ThendJyc）
- 在第三方登录密码错误时会自动退出登录（#5090，@LeiSureLyYrsc）
- 在点击 Mod 更新图标后会弹窗确认，以免误触导致 Mod 被更新（#5189，@Lmy1503）
- 更新资源下载的筛选列表（#5212，PR @3gf8jv4dv）
- 更换帮助库图床（LTCatt/PCL2Help#397，PR @z0z0r4）
- 将 [简化 Mod 加载器显示] 设置改为 [显示 Mod 加载器时忽略 Quilt]
    开启该选项时，不再隐藏 NeoForge 相关的信息
    开启该选项时，Mod 详情页面也不会单独展示 Mod 所支持的 Quilt 版本了
- 更新 11 月赞助者列表，感谢大家对 PCL 的支持！
- 由于特定权限下无法拖拽文件到 PCL 窗口，Mod 管理的安装 Mod 选项不再只给出“可以通过拖拽来安装 Mod”的提示，而是确实可以选择文件并安装了
- 部分音乐播放失败时，会将播放失败的音乐自动移出播放列表
- 如果尝试一次性更新 15+ 个 Mod，会弹窗提示更新 Mod 可能带来的风险
- 将 -Xmn 的比例从 15% 调整为 20%
- 更新 Mod 数据库

### Bug 修复
- 在特定情况下切换主页会出现额外的报错（#3270）
- 更多页面的左边栏选项没有对齐（#3491，@azaz-az）
- SafeList 的 Linq 调用不是线程安全的，会导致随机偶发的错误（#3771，@WorldHim）
- 指向拖动条时可能导致意外的页面滚动（#3854，@lactobionicAcid）
- 可能无法打开 PCL，提示“PCL 初始化错误”（#4517，@Immortal-Sty）
- 使用部分低版本 Java 启动 MC 1.12.2- 时窗口过小（#4654，@dsb987）
- [简化 Mod 加载器显示] 的提示不准确（#4968，@dd2024is2013）
- MyImage 无法加载部分网站的图片（#5072，@Joker2184，PR @tangge233）
- 部分音乐出错导致启动器出错（#5096，@Gluoluo，PR @lactobionicAcid）
- 32 位系统下可能无法启动游戏（#5099，@SongzqInChina，由 @00ll00 修复）
- 过长的 Mod 名称会遮挡更新按钮（#5101，@QianFuv）
- 无法取消版本设置中的版本隔离选项修改提示（#5118，@LingXu268）
- 在未安装 Java 时提示不推荐安装 32 位 Java（#5128，PR @Hill23333）
- 第三方登录在退出后不会重新尝试登录（#5131，@wuliaodexiaoluo）
- 刷新 Mod 列表后不会显示前置 Mod（#5173，@WinExp）
- 联网更新主页的下载地址无法被清空（#5182，@newbeflash）
- 当镜像源未更新新版 MC 时，可能无法安装相应的整合包（#5195，@tucaoba2333）
- 当镜像源未更新新版 Forge 时，可能无法安装相应的 Forge（#5215，@duzefang）
- 启动前执行命令 Tooltip 中存在笔误（#5225，PR @Ad-closeNN）
- 选择“尽量使用官方源”时下载失败，可能会永远卡在加载中，而不会显示失败（#5232，@songyy0415）
- 赞助弹窗和购买正版弹窗会在同一次启动中一起弹出
- 刷新 Mod 列表时不会刷新 Mod 详情中的文件列表

### 后续计划
- #4758

<hr /><em>This discussion was created from the release <a href='https://github.com/Hex-Dragon/PCL2/releases/tag/2.8.11'>2.8.11</a>.</em>

**Comment by LTCatt** [comment]
已重新发布该更新以修复 #5238

**Comment by LinQingYuu** [comment]
龙猫你忘记标 Pre-release 了（这个 Latest 害的我以为正式版也更新了）

Edit:(兴冲冲跑去更新结果发现最新版本还是 2.8.9)

**Reply by LTCatt** [reply]
如果预计会在数天后发布正式版就会标 Release，不计划发布正式版就是 Pre（

**Reply by LinQingYuu** [reply]
> 如果预计会在数天后发布正式版就会标 Release，不计划发布正式版就是 Pre（

wai，原来是这样吗

**Reply by Joker2184** [reply]
> > 如果预计会在数天后发布正式版就会标 Release，不计划发布正式版就是 Pre（
> 
> wai，原来是这样吗

原来如此+1
成功把更新日志自动更新写炸了.jpg
跑完checkversion发现咋是正式版的图片 ;D

**Reply by allMagicNB** [reply]
2.8.10 呢（

**Reply by Joker2184** [reply]
> 2.8.10 呢（

啥（ 还没加（

明天再说.jpg 在研究action.jpg

**Reply by tangge233** [reply]
> 2.8.10 呢（

此版本没有推送到正式版本，因为发现了更多的 Bug…
